uint8[] key
---
bool set
